code/modules/reagents/reagent_containers/food/drinks/juicebox.dm

```diff
@@ -16,32 +68,60 @@
 /obj/item/chems/food/drinks/juicebox/open(mob/user)
 	playsound(loc,'sound/effects/bonebreak1.ogg', rand(10,50), 1)
 	to_chat(user, SPAN_NOTICE("You pull off the straw and stab it into \the [src], perforating the foil!"))
-	icon_state = "[initial(icon_state)]_open"
 	atom_flags |= ATOM_FLAG_OPEN_CONTAINER
+	appearance = update_juicebox()
```

Comment:
Probably should just call `update_icon()` here since you assign the new appearance there anyway.